test/unit/BPool/BPool_Bind.tree

```diff
@@ -0,0 +1,29 @@
+BPool::Bind
+├── when reentrancy lock is set
+│   └── it should revert
+├── when caller is NOT controller
+│   └── it should revert
+└── when caller is controller
+    ├── when token is already bound
+    │   └── it should revert
+    ├── when pool is finalized
+    │   └── it should revert
+    ├── when MAX_BOUND_TOKENS tokens are already bound
+    │   └── it should revert
+    ├── when token weight is too low
+    │   └── it should revert
+    ├── when token weight is too high
+    │   └── it should revert
+    ├── when too little balance is provided
+    │   └── it should revert
+    ├── when weight sum exceeds MAX_TOTAL_WEIGHT
+    │   └── it should revert
+    └── when token can be bound
+        ├── it reads the reentrancy lock
```

Comment:
wdym it "reads"? i think this is being tested on `when reentrancy lock is set`, i wouldn't add this expectation

Reply:
addressing in #140